### ISSUE TITLE
implicit 0's behaviour-change

### ISSUE THREAD

**Issue by valentinradu** [spencermountain/spacetime#90]
`spacetime('2019-01-25T20:00:00+01:00').isBefore('2019-01-25T20:00:00+01:00')` returns `true`, but clearly is the same date. Prob is, when milliseconds are not specified, the parser doesn't assume 0, but a random (prob taken from current date) value.

**Comment by spencermountain** [comment]
fixed in `5.2.2`.

yeah, I don't mind the idea of using implicit values from the current time - 
if you say `spacetime('june 5th 2018')` i don't mind setting the current hour, minute, etc.

Those are the same assumptions made in  js date object.

But if this generates race-conditions, like your example, we should avoid those, at least.

I've co-erced millisecond, and second to 0, unless they're specified.
Minutes and hours will be left unchanged. 

It's possible that if you run this on the last millisecond before a minute changes, it'll happen again.

¯\_(ツ)_/¯
thanks

**Comment by hunteva** [comment]
Thanks for the fix @spencermountain but I'm afraid js date doesn't make that assumptions.
```
> new Date('June 5 2018')
2018-06-04T12:00:00.000Z
> new Date('2018-06-06')
2018-06-06T00:00:00.000Z
> new Date(2018, 05, 05)
2018-06-04T12:00:00.000Z
> new Date()
2019-02-05T01:42:51.432Z
```
except there is no arguments passed in, feels it might be nicer to have the same behaviour as js date object, I could make those changes if you like.
